src/language/walker/programAwareRuleWalker.ts

```diff
@@ -20,8 +20,8 @@ import {IOptions} from "../../lint";
 import {RuleWalker} from "./ruleWalker";
 
 export class ProgramAwareRuleWalker extends RuleWalker {
-    protected program: ts.Program;
-    protected typeChecker: ts.TypeChecker;
+    private program: ts.Program;
```

Comment:
historically we've tended towards that code style so it's easy to locate all the fields in one place. we've loosened up the guidelines a bit; I don't really mind using constructor param properties.